Presentation/UINavigationController+Presenting.swift

```diff
@@ -26,6 +26,13 @@ public extension PresentationOptions {
 extension UINavigationController: PresentingViewController {
     public func present(_ vc: UIViewController, options: PresentationOptions) -> PresentingViewController.Result {
         let dismissFuture = vc.installDismissButton().future.map { throw PresentError.dismissed }
+
+        if options.contains(.prefersNavigationBarHidden) {
+            self.isNavigationBarHidden = true
+        } else if options.contains(.prefersNavigationBarShown) {
+            self.isNavigationBarHidden = false
+        }
+        
```

Comment:
So what happens if we push another controller with none of the above options set? Will the navigation bar be hidden or not in that case or rather be dependent on the previous pushed view controller? Are the "non of options above" a way to allow users setting `isNavigationBarHidden` explicitly form the out-side. What if we say that setting `isNavigationBarHidden` from the out-side is not supported, would it make sense to only have `.prefersNavigationBarHidden` and change the above to:

```swift
isNavigationBarHidden = options.contains(.prefersNavigationBarHidden)
```

Reply:
The state of `isNavigationBarHidden` remains unchanged until the preference option is specified in the `PresentationOptions` of next presentation.

If we think of UIKit way of doing it, we can always instantiate `UINavigationController` and hide the `UINavigationBar` on the `UINavigationController` object. And when a `UIViewController` object wants to hide it, we simply access `UIViewController` objects `navigationController` property. But that is usually done in the life cycle methods (after the ViewController is actually in the navigation stack). But as we do not play around with the UIViewController view's life-cycle methods, we felt presentation option is the way to go.

`isNavigationBarHidden = options.contains(.prefersNavigationBarHidden)`

This is the way we tried it before. But we realised, that it will be breaking the scenarios, pre-existent in the app. As we will have to specify the option everywhere, based on the preference. Hence we check for explicit specification of the controllers preference.

Reply:
So are we currently restoring anything after a view controller has been displayed using these new options? I wonder if we should keep a state in the `PushPoper` helper of a view controllers's preferred navigation bar visibility and use that to restore when it becomes visible again such as after a pop?

Reply:
I still would like to investigate if we can just settle on one `.prefersNavigationBarHidden` options but still make it work together with external changes. Not sure when an externally update is typically set. If it is set in materialize, we typically don't have a navigation controller available until perhaps we get the install callback. So if we store the preferred state based on options in PushPoper when being pushed, and then if the navigation bar visibility is externally changed while being presented and the view controller later is moved out of view (another controller is pushed), we can perhaps updated its PushPoper state if necessary. 

Could you elaborate the problems you currently hare having with mixing the options and external update?

Reply:
We have added an example in the example project. That is typically the scenario we are trying to support.

ViewController1(Navbar visible) -> ViewController2(Navbar hidden) -> ViewController3(Navbar visible)